libraries/standard/mqtt/cbmc/stubs/get_time_stub.c

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "mqtt.h"
+#include "get_time_stub.h"
+
+uint32_t GetCurrentTimeStub( void )
+{
+    /* There exists loops in the MQTT library that rely on the timestamp being
+     * reasonable in order to complete. Returning an unbounded timestamp does
+     * not add value to the proofs as the MQTT library uses the timestamp for
+     * only arithmetic operations. In C arithmetic operations on unsigned
+     * integers are guaranteed to reliably wrap around with no adverse side
+     * effects. */
```

Comment:
I see this comment, but is there any value in time advancing in nondeterministic, positive values?  And not just by 1?

Reply:
I don't think so. The time is used just for how often to loop and by advancing non-deterministically we could loop for a very long time.

Reply:
I will add this reasoning to the comments.